fuel/iterator.py

```diff
@@ -23,9 +23,15 @@ def __iter__(self):
 
     def __next__(self):
         if self.request_iterator is not None:
-            data = self.data_stream.get_data(next(self.request_iterator))
+            if self.data_stream.batch :
```

Comment:
Mmmm, maybe it's better to move this logic to the `Transformer` base class. That way child classes can override it if necessary and don't need to duplicate methods if the implementation for batch/example is the same.